R/estimateDiversity.R

```diff
@@ -178,13 +188,123 @@ setMethod("estimateDiversity", signature = c(x = "SummarizedExperiment"),
                                          .get_diversity_values,
                                          x = x,
                                          mat = assay(x, abund_values),
+                                         tree = tree,
                                          BPPARAM = BPPARAM,
                                          ...)
         .add_values_to_colData(x, dvrsts, name)
     }
 )
 
+#' @rdname estimateDiversity
+#' @export
+setMethod("estimateDiversity", signature = c(x="TreeSummarizedExperiment"),
+    function(x, abund_values = "counts",
+             index = c("shannon","gini_simpson","inverse_simpson",
+                       "coverage", "fisher", "faith"),
+             name = index, ..., BPPARAM = SerialParam()){
+        
+        # Gets the tree 
+        tree <- rowTree(x)
```

Comment:
The implementation almost works now. However, since you don't define `tree` for the `SummarizedExperiment` method, line 191 will not work. If you just remove it, it will work again.

Reply:
removed